packages/modals/src/styled/StyledModal.ts

```diff
@@ -6,35 +6,107 @@
  */
 
 import PropTypes from 'prop-types';
-import styled from 'styled-components';
-import classNames from 'classnames';
-import { retrieveComponentStyles, isRtl, DEFAULT_THEME } from '@zendeskgarden/react-theming';
-import ModalStyles from '@zendeskgarden/css-modals';
+import styled, { css, keyframes, ThemeProps, DefaultTheme } from 'styled-components';
+import { getColor, retrieveComponentStyles, DEFAULT_THEME } from '@zendeskgarden/react-theming';
+import { StyledFooter } from '../styled';
 
-const COMPONENT_ID = 'modals.modal_view';
+const COMPONENT_ID = 'modals.modal';
 
 export interface IStyledModalProps {
   isLarge?: boolean;
+  isCentered?: boolean;
   isAnimated?: boolean;
 }
 
+const animationStyles = () => {
+  const animationName = keyframes`
+    0% {
+      transform: scale(0);
+      opacity: 0;
+    }
+
+    50% {
+      transform: scale(1.05);
+    }
+
+    to {
+      opacity: 1;
+    }
+  `;
+
+  return css`
+    animation-name: ${animationName};
+    animation-duration: 0.3s;
+    animation-timing-function: ease-in-out;
+  `;
+};
+
+const boxShadow = (props: ThemeProps<DefaultTheme>) => {
+  const { theme } = props;
+  const { space, shadows } = theme;
+  const offsetY = `${space.base * 5}px`;
+  const blurRadius = `${space.base * 7}px`;
+  const color = getColor('neutralHue', 800, theme, 0.35);
+
+  return shadows.lg(offsetY, blurRadius, color as string);
+};
+
+const largeFooterStyles = () => css`
+  ${StyledFooter} {
+    border: ${props => `${props.theme.borders.sm} ${getColor('neutralHue', 200)}`};
+    padding: ${props => `${props.theme.space.base * 8}px ${props.theme.space.base * 10}px`};
+  }
+`;
+
 /**
- * Accepts all `<div>` props
+ * 1. IE11 centering hack.
  */
-export const StyledModal = styled.div.attrs<IStyledModalProps>(props => ({
+export const StyledModal = styled.div.attrs<IStyledModalProps>({
   'data-garden-id': COMPONENT_ID,
-  'data-garden-version': PACKAGE_VERSION,
-  className: classNames(ModalStyles['c-dialog'], {
-    // Sizing
-    [ModalStyles['c-dialog--large']]: props.isLarge,
-
-    // States
-    [ModalStyles['is-open']]: props.isAnimated,
-
-    // RTL
-    [ModalStyles['is-rtl']]: isRtl(props)
-  })
-}))<IStyledModalProps>`
+  'data-garden-version': PACKAGE_VERSION
+})<IStyledModalProps>`
+  display: flex;
+  position: fixed;
+  flex-direction: column;
+  margin: ${props => (props.isCentered ? '0' : `${props.theme.space.base * 12}px`)};
+  border: ${props => `${props.theme.borders.sm} ${getColor('neutralHue', 300)}`};
+  border-radius: ${props => props.theme.borderRadii.md};
+  border-color: transparent;
+  box-shadow: ${boxShadow};
+  background-color: ${props => props.theme.colors.background};
+  width: ${props => (props.isLarge ? '800px' : '542px')};
+  min-height: 60px;
+  max-height: calc(100vh - 96px);
```

Comment:
Computing `60px` with `space.base` wasn't used because a minor adjustment in `space.base` may result in a very large `min-height` value. I also left the `calc` in here, as I was unsure what the CSS-in-JS equivalent is.

If `max-height` needs to be changed, I would love some suggestions, and the steps to get this value right.